test/Test/Pos/Lrc/WorkerSpec.hs

```diff
@@ -0,0 +1,264 @@
+        let genSmallRichStake =
+                unsafeAddCoin thresholdCoin . mkCoin <$> choose (0, 10)
+        richStake1 <- genSmallRichStake
+        richStake2 <- genSmallRichStake
+        let richStake3 =
+                foldl'
+                    unsafeSubCoin
+                    totalStakeGroup
+                    [poorStake, richStake1, richStake2]
+        let stakes = [poorStake, richStake1, richStake2, richStake3]
+        case richStake3 >= thresholdCoin of
+            True  -> return (addresses, CustomStakes stakes)
+            False -> error "threshold is too big, tests are not ready for it"
+
+lrcCorrectnessProp :: BlockProperty ()
+lrcCorrectnessProp = do
+    let k = Const.blkSecurityParam
+    -- This value is how many blocks we need to generate first. We
+    -- want to generate blocks for all slots which will be considered
+    -- in LRC except the last one, because we want to include some
+    -- special transactions into it.  We don't use 'crucialSlot' or
+    -- anything similar, because we don't want to rely on the code,
+    -- but rather want to use our knowledge.
+    let blkCount0 = 8 * k - 1
+    () <$ bpGenBlocks (Just blkCount0) (EnableTxPayload False) (InplaceDB True)
+    genAndApplyBlockFixedTxs =<< txsBeforeBoundary
+    -- At this point we have applied '8 * k' blocks. The current state
+    -- will be used in LRC.
+    stableUtxo <- lift GS.getAllPotentiallyHugeUtxo
+    stableStakes <- lift GS.getAllPotentiallyHugeStakesMap
+    -- All further blocks will not be considered by LRC for the 1-st
+    -- epoch. So we include some transactions to make sure they are
+    -- not considered.
+    genAndApplyBlockFixedTxs =<< txsAfterBoundary
+    -- We need to have at least 'k' blocks after the boundary to make
+    -- sure that stable blocks are indeed stable. Note that we have
+    -- already applied 1 blocks, hence 'pred'.
+    blkCount1 <- pred <$> pick (choose (k, 2 * k))
+    () <$ bpGenBlocks (Just blkCount1) (EnableTxPayload False) (InplaceDB True)
+    lift $ Lrc.lrcSingleShotNoLock 1
+    leaders1 <-
+        maybeStopProperty "No leaders for epoch#1!" =<< lift (Lrc.getLeaders 1)
+    -- Here we use 'genesisSeed' (which is the seed for the 0-th
+    -- epoch) because we have a contract that if there is no ssc
+    -- payload the previous seed must be reused (which is the case in
+    -- this test).
+    let expectedLeadersUtxo =
+            Lrc.followTheSatoshiUtxo Lrc.genesisSeed stableUtxo
+    let expectedLeadersStakes =
+            Lrc.followTheSatoshi Lrc.genesisSeed (HM.toList stableStakes)
+    unless (expectedLeadersUtxo /= leaders1) $
+        stopProperty "expectedLeadersUtxo /= leaders1"
+    unless (expectedLeadersStakes /= leaders1) $
+        stopProperty "expectedLeadersStakes /= leaders1"
+    checkRichmen
+
+checkRichmen :: BlockProperty ()
+-- Here we check richmen.  The order must be the same as the one
+-- in 'allRichmenComponents'. Unfortunately, I don't know how to
```

Comment:
Because when we generate genesis, we use first 4 secret keys for the first group (dedicate to the first richmen component), next 4 secret keys for the second group, etc.